src/App.js

```diff
@@ -24,7 +24,24 @@ const App = () => {
       setSquares[index] = current
       // Adding 1 to count to switch turns 
       setTurn(turn+1)
+    
+  }
+}
+useEffect(()=>{
+checkTurn()
+
+},[squares, turn])
```

Comment:
useEffect can be super helpful and essentially will run anytime React recognizes a change in state.  This particular problem could be approached with both useEffect, or checking the turn upon each click.

Reply:
useEffect ended up solving our problem here but we are going to experiment with what you mean with checking the turn upon each click! Thank you!